### PR TITLE
fix: OE UI polish

### PR DESCRIPTION
**Pull request by fPolic**
**What**
- fix bugs, align implementations of the modal and the timeline, fetch optimisations

**TODO**
- [x] use the list endpoint instead of `edits`relation (FIXES CORE-697)
- [x] hide timeline event while editing and handle stale edits (FIXES CORE-688)
- [x] refund OE doesn't show refund time line event for refunding (FIXES CORE-692)
- [x] OE with removed line items shows incorrectly (FIXES CORE-690)
- [x] add spacing in order edit time line event (FIXES CORE-708)
- [x] "Customer payment required" should not be shown if difference due is 0 (FIXES CORE-707)
- [x] copy SKU in edit modal (FIXES CORE-709)
- [x] Timeline event with "by" field should show email when no name is available (FIXES CORE-691)
- [x] show prices with taxes in modal
- [x] allow operators to select variants despite being out of stock
- [x] confirmation-request sent should show based on Notifications (FIXES CORE-718)
- [x] "Spamming" the quantity ticker on order editing results in error (FIXES CORE-717)

### CODE REVIEW

**Comment by srindom** [comment]
When I filter in the variants table it doesn't show the changes as I type; the table seems to update though

**Comment by srindom** [comment]
thought(non-blocking,if-minor): maybe we should include quantities when we remove/modify items e.g. if you reduce a LineItem right now it will be listed in the timeline as "removed". Wdyt?

**Comment by srindom** [comment]
Really great work @fPolic !! 💪

**Comment by srindom** [comment]
We should update the tooltip when a line item has `isLocked === true` to say:

This line item is part of a fulfillment and cannot be edited. Cancel the fulfillment to edit the line item.

File: src/components/molecules/timeline-events/order-edit/confirmed-difference-due.tsx [hunk 1]
```diff
@@ -1,50 +1,70 @@
-import { useAdminOrderEdit } from "medusa-react"
-import React from "react"
+import { useAdminOrder, useAdminOrderEdit } from "medusa-react"
```
Comment (srindom) [inline src/components/molecules/timeline-events/order-edit/confirmed-difference-due.tsx:1]: general thought on this one: 

I think we should consider showing the refund timeline event regardless of any OrderEdits. The approach would be this:

There is always a check for `order.paid_total > order.total` if the check is >0 then we show the refund timeline event. 

The narrative here is: 
We don't care how this discrepancy between paid and actual total arose, but we owe the customer money so now we should issue a refund. 

I will move this thought to an issue that we can consider separately, justed wanted to note it here.

File: src/components/molecules/timeline-events/order-edit/requested.tsx [hunk 2]
```diff
@@ -1,23 +1,61 @@
...
 const EditRequested: React.FC<RequestedProps> = ({ event }) => {
+  const [showResend, setShowResend] = useState(false)
+
+  const { notifications } = useAdminNotifications({
+    resource_id: event.edit?.id,
+  })
+
+  const notification = notifications?.find(
+    (n) => n.event_name === "order-edit.requested"
+  )
```
Comment (srindom) [inline src/components/molecules/timeline-events/order-edit/requested.tsx:23]: praise: brilliant!

File: src/components/molecules/timeline-events/order-edit/confirmed-difference-due.tsx [hunk 3]
```diff
@@ -1,50 +1,70 @@
...
+        <Button
+          onClick={() => setShowRefundModal(true)}
+          variant="ghost"
+          size="small"
+          className="w-full border border-grey-20 mb-xsmall text-rose-50"
+        >
+          Refund
           {formatAmountWithSymbol({
-            amount: orderEdit.difference_due,
+            amount: Math.abs(order?.total - order?.paid_total),
```
[the remark this answers is not in the document]
Comment (srindom) [reply src/components/molecules/timeline-events/order-edit/confirmed-difference-due.tsx:54]: realizing that we need to take refunded total into account. So should be `order.paid_total - order.refunded_total - order.total`

**Review by srindom — COMMENTED** [review]
Really good work! Have a few small nits and then we are good to go!

File: src/components/molecules/timeline-events/order-edit/confirmed-difference-due.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (fPolic) [reply src/components/molecules/timeline-events/order-edit/confirmed-difference-due.tsx]: changed [here](https://github.com/medusajs/admin/pull/737/commits/1340a23b44e303223e182ad3ab9f5cfaf8445ec6)

File: src/components/molecules/timeline-events/order-edit/confirmed-difference-due.tsx (hunk 1 above)
Comment (fPolic) [reply src/components/molecules/timeline-events/order-edit/confirmed-difference-due.tsx]: Added a generic timeline event for refunds that is rendered always on top of the list if there is refundable amount in the last commit, let me know what you think :)

**Review by srindom — APPROVED** [review]
LGTM!

**Pull request merged**